### PR TITLE
version: Start 8.5.0dev

### PR DESCRIPTION
**Pull request by wenzeslaus**
Update the version file and update hardcoded version numbers.

* Update hardcoded version in Docker files.
* Update hardcoded version in Windows CI.
* Replace hardcoded version by query to grass executable in Linux CI.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -277,20 +277,20 @@ RUN make distclean || echo "nothing to clean"
...
-    rm -rf /usr/local/grass84/share; \
-    mkdir -p /usr/local/grass84/gui/wxpython/xml/; \
-    mv module_items.xml /usr/local/grass84/gui/wxpython/xml/module_items.xml;
+    &&  cp /usr/local/grass85/gui/wxpython/xml/module_items.xml module_items.xml; \
+    rm -rf /usr/local/grass85/demolocation; \
+    rm -rf /usr/local/grass85/fonts; \
+    rm -rf /usr/local/grass85/gui; \
+    rm -rf /usr/local/grass85/share; \
+    mkdir -p /usr/local/grass85/gui/wxpython/xml/; \
+    mv module_items.xml /usr/local/grass85/gui/wxpython/xml/module_items.xml;
```
Comment (echoix) [inline Dockerfile:286]: We could all replace them with a simple build ARG. What should it be called?

But in a docker image, could we assume a versionless path, as only one version is shipped in it? What is coupled to that path?
Below there is already a link that creates a versionless path.

**Comment by echoix** [comment]
I've been playing around advanced custom managers within renovate, and I know that it is possible to also create replacements (ie to migrate to another dependency). Either with or without a preceding/leading marker comment, it would be possible to use renovate to finish replacing the unavoidable version changes. 

If I find how, maybe the version file could be used as the source, otherwise the config file would need to be adapted too :s. 
In all cases, the update would be applied as a second PR after the change of the version. It isn't perfect..

File: Dockerfile (hunk 1 above)
Comment (wenzeslaus) [reply Dockerfile:286]: Please, follow on that in #3791. I don't plan to address that in this PR.

**Comment by wenzeslaus** [comment]
> ...Either with or without a preceding/leading marker comment, it would be possible to use renovate to finish replacing the unavoidable version changes.

There is definitively couple at this point which are unavoidable and comment may help. It might be possible to mark these with a comment and use that comment to find them even without renovate. Figuring out how to consistently create a comment for all these situations might be more valuable than the actual automation of the update. Now you are looking though a grep result and trying to find out what are actually relevant instances. Now there is much less, so it's easier, but could be better.

**Comment by echoix** [comment]
In that case, you might be interested in the concepts used by the tool called release-please. It works for semantic commits, so it isn't adapted as such here, but the ideas are there.
Super-linter has traces of it even in their README.

File: Dockerfile (hunk 1 above)
Comment (neteler) [inline Dockerfile:286]:
```suggestion
    &&  cp /usr/local/grass8?/gui/wxpython/xml/module_items.xml module_items.xml; \
    rm -rf /usr/local/grass8?/demolocation; \
    rm -rf /usr/local/grass8?/fonts; \
    rm -rf /usr/local/grass8?/gui; \
    rm -rf /usr/local/grass8?/share; \
    mkdir -p /usr/local/grass8?/gui/wxpython/xml/; \
    mv module_items.xml /usr/local/grass8?/gui/wxpython/xml/module_items.xml;
```

... maybe even easier?

File: Dockerfile [hunk 2]
```diff
@@ -277,20 +277,20 @@ RUN make distclean || echo "nothing to clean"
...
 
 # Build the GDAL-GRASS plugin
 RUN git clone https://github.com/OSGeo/gdal-grass \
     && cd "gdal-grass" \
     && ./configure \
       --with-gdal=/usr/bin/gdal-config \
-      --with-grass=/usr/local/grass84 \
+      --with-grass=/usr/local/grass85 \
```
Comment (neteler) [inline Dockerfile:293]:
```suggestion
      --with-grass=/usr/local/grass8? \
```

File: Dockerfile (hunk 1 above)
Comment (neteler) [reply Dockerfile:286]: Or we do that elsewhere.

Comment (echoix) [reply Dockerfile:286]: It is possible to use a wildcard when creating a directory with mkdir?

Comment (echoix) [reply Dockerfile:286]: > Or we do that elsewhere.

Let's add this to a small todo for this week

**Comment by echoix** [comment]
Let's start 8.5.0 officially!

**Pull request merged**